src/cryptonote_core/blockchain.cpp

```diff
@@ -419,6 +420,10 @@ bool Blockchain::init(BlockchainDB* db, const network_type nettype, bool offline
     // ensure we fixup anything we found and fix in the future
     m_db->fixup();
   }
+  else if (m_regtest)
+  {
+    m_db->set_batch_transactions(true);
+  }
```

Comment:
The reg test wouldn’t work if it was omitted IIRC. The core tests also has this call.

Reply:
That seems odd since there are very few differences between non-core-tests fakechain and mainnet. In any case, I asked hyc whether batch transactions can be enabled by default.

Reply:
hyc sees no reason why they should not be enabled by default. Let it be so :)

Reply:
Shall I:
- remove the "if (m_regtest)"
- remove "set_batch_transactions(true);" in the other locations: 
blockchain_db.cpp/db_lmdb.cpp/blockchain_import.cpp/hardfork.cpp (and chaingen.h if I don't break anything) ?

Reply:
I'd remove that whole if (regtest) { batch; } thing, and default batch transactions to true wherever they're defined (db_lmdb.cpp probably). That is, probably set a bool to true in the ctor, wahtever it is the function does.

Reply:
I've pushed my interpretation of @moneromooo-monero's suggestion to my updated branch: https://github.com/stoffu/monero/commits/regtest